### PR TITLE
lwm2m_client: Replace GPS driver with GNSS interface

### PR DESCRIPTION
**Pull request by jarlamsa**
Replace the deprecated GPS driver with the new GNSS interface.
Currently the A-GPS isn't supported, because of that waiting for a fix might take some time as the periodic LTE wakeups are interfering with the GNSS-module.

@tokangas by default the lwm2m_client contacts the lwm2m server every ~25 seconds, which quite effectively halts the progress of using the tracked satellites. The usage of A-GPS will probably solve that issue.

### CODE REVIEW

**Review by SeppoTakalo — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -192,6 +192,7 @@ nRF9160 samples
 
     * Reworked the retry logic so that the sample can fall back to bootstrap mode and need not always restart the LTE connection.
       Added :ref:`state_diagram` to the sample documentation.
+    * Replaced the deprecated GPS-driver with the new GNSS interface
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:195]:
```suggestion
    * Replaced the deprecated GPS-driver with the new GNSS interface.
```

**Review by umapraseeda — COMMENTED** [review]
Any updates required to the LwM2M client Readme with regards to the replacement of GPS driver?

File: samples/nrf9160/lwm2m_client/src/sensors/gnss_module.c [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+static struct nrf_modem_gnss_pvt_data_frame pvt_data;
+
+/**@brief Callback for GNSS events */
+static void gnss_event_handler(int event_id)
+{
+	int err = 0;
+
+	switch (event_id) {
+	case NRF_MODEM_GNSS_EVT_PVT: {
+		err = nrf_modem_gnss_read(&pvt_data, sizeof(pvt_data), NRF_MODEM_GNSS_DATA_PVT);
```
Comment (tokangas) [inline samples/nrf9160/lwm2m_client/src/sensors/gnss_module.c:26]: This code could be moved to case `NRF_MODEM_GNSS_EVT_FIX` instead. It gets the same PVT data, so you could just remove the `pvt_data.flags & NRF_MODEM_GNSS_PVT_FLAG_FIX_VALID` check.

File: samples/nrf9160/lwm2m_client/src/sensors/gnss_module.c [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	if (err) {
+		LOG_ERR("Could not set fix interval for GNSS %d (%d)", search_interval, err);
+	}
+
+	uint16_t search_timeout = CONFIG_GNSS_SEARCH_TIMEOUT_TIME;
+
+	err = nrf_modem_gnss_fix_retry_set(search_timeout);
+
+	if (err) {
+		LOG_ERR("Could not set fix timeout for GNSS %d (%d)", search_timeout, err);
```
Comment (tokangas) [inline samples/nrf9160/lwm2m_client/src/sensors/gnss_module.c:85]: And here.

**Review by tokangas — COMMENTED** [review]
Looks pretty good to me. Have you though about enabling eDRX or PSM for this sample? Those would enable GNSS to get fixes also without GNSS priority or A-GPS.

I noticed that there were a few new/renamed files with copyright year 2021. For new files it should at least be 2022, but I'm not sure how it should be here because I think that all files have a predecessor for the GPS driver interface, although GitHub shows some as new and some as renamed.

File: samples/nrf9160/lwm2m_client/src/sensors/gnss_module.c [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	}
+
+#if defined(CONFIG_GNSS_PRIORITY_ON_FIRST_FIX)
+	/* Priority mode is disabled automatically after first fix */
+	err = nrf_modem_gnss_prio_mode_enable();
```
[the remark this answers is not in the document]
Comment (jarlamsa) [reply samples/nrf9160/lwm2m_client/src/sensors/gnss_module.c:90]: Good point. The registration can happen again if for some reason the connection between client and server closes and the client needs to re-register. Changed the behaviour a bit now.

**Comment by jarlamsa** [comment]
Thank you for your reviews @umapraseeda @tokangas @rlubos I think I have addressed your review-comments.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -207,6 +207,7 @@ nRF9160 samples
 
     * Reworked the retry logic so that the sample can fall back to bootstrap mode and need not always restart the LTE connection.
       Added :ref:`state_diagram` to the sample documentation.
+    * Replaced the deprecated GPS-driver with the new GNSS interface.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:210]:
```suggestion
    * Replaced the deprecated GPS driver with the new GNSS interface.
```

File: samples/nrf9160/lwm2m_client/README.rst [hunk 6]
```diff
@@ -369,15 +369,15 @@ CONFIG_APP_TEMP_SENSOR - Configuration for enabling an LwM2M Temperature sensor
...
-CONFIG_APP_GPS - Configuration for enabling GNSS functionality
+CONFIG_APP_GNSS - Configuration for enabling GNSS functionality
    This configuration might interfere with LTE if the GNSS conditions are not optimal.
    Disable this option if GNSS is not needed.
 
-.. _CONFIG_GPS_PRIORITY_ON_FIRST_FIX:
+.. _CONFIG_GNSS_PRIORITY_ON_FIRST_FIX:
 
-CONFIG_GPS_PRIORITY_ON_FIRST_FIX - Configuration for prioritizing GNSS over LTE during the search for first fix.
+CONFIG_GNSS_PRIORITY_ON_FIRST_FIX - Configuration for prioritizing GNSS over LTE during the search for first fix.
```
Comment (umapraseeda) [inline samples/nrf9160/lwm2m_client/README.rst:380]:
```suggestion
CONFIG_GNSS_PRIORITY_ON_FIRST_FIX - Configuration for prioritizing GNSS over LTE during the search for first fix
```

**Review by umapraseeda — APPROVED** [review]
Some more minor fixes..otherwise LGTM!

**Review by rlubos — APPROVED** [review]

File: samples/nrf9160/lwm2m_client/src/events/gnss_pvt_event.c [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+#include "gnss_pvt_event.h"
+
+#include <stdio.h>
+
+static void log_gnss_event(const struct event_header *eh)
+{
+	struct gnss_pvt_event *event = cast_gnss_pvt_event(eh);
+
+	EVENT_MANAGER_LOG(eh, "gnss_pvt_event lat: %lf long: %lf alt: %lf",
```
Comment (tokangas) [inline samples/nrf9160/lwm2m_client/src/events/gnss_pvt_event.c:15]: Nitpicking, but `event->pvt.altitude` is a `float`, so `%f` is enough.

**Review by tokangas — APPROVED** [review]

**Pull request merged**